lib/routes/email.js

```diff
@@ -17,14 +22,25 @@ module.exports = {
     }
   },
   handler: function email(req, reply) {
-    var email = req.auth.credentials.email;
-    if (email) {
+    request.get(AUTH_SERVER_URL, {
+      headers: {
+        Authorization: 'Bearer ' + req.auth.token
+      },
+      json: true
```

Comment:
Yes, it will. There's no real way around it as long as the auth server has the email, and the profile server has other details.